sentinelsat/sentinel.py

```diff
@@ -318,28 +315,42 @@ def download(self, id, directory_path='.', checksum=False, check_existing=False)
 
         self.logger.info('Downloading %s to %s' % (id, path))
 
-        # Check if the file exists and passes md5 test
-        # The download function will by default continue the download if the file exists but is incomplete
-        if exists(path) and getsize(path) == product_info['size']:
-            if not check_existing or _md5_compare(path, product_info['md5']):
-                self.logger.info('%s was already downloaded.' % path)
-                return product_info
+        if exists(path):
```

Comment:
Agreed. After thinking about it I do support your proposition
> Add a separate method for this (checksum()?) functionality, which takes a list of ids as input.

Local checksum tests are not a `download` but rather a separate function, that I would expect to call with either a folder (using the filenames as reference), a list of product IDs or a full OData dictionary. I suspect this would then be a but much for this PR - so I'd say you leave out `check_existing` in this PR and we'll deal with it in another Issue/PR.